### DISCUSSION TITLE
[RFC] 037 - LobeChat 1.0 Milestone

### DISCUSSION THREAD

**Discussion by arvinxx** [lobehub/lobehub#1890] [RFCs]
## 背景

- https://github.com/lobehub/lobe-chat/discussions/1768

本篇为 RFC

## 核心设计

结合 LobeChat 过去半年来的用户反馈，1.0 会从以下几个方面进行特性研发:

- [x] 服务端数据库： #1851 ，这个不细说了，万众期待，做就完事
- [x] 语言模型选择器: #1686 ，应该可以解决掉 95% 和模型选择、配置的需求和体验问题。（基本上把 provider 的实现代码优化了一版）
- [ ] ~~Uniform Provider: 兼容与 liteLLM/one-api/openrouter 这类 ai gateway 的集成问题，便于更统一管理，同时仍然获得体验良好的选择面板。~~ 在选择器基础上，感觉这个可能暂时不需要了
- [x] Feature Flag: #1894 有很多用户在做私有化部署时会有各种隐藏特性的需求。我们一直没做一个系统化的方案。本次会计划提供一种 feature flag 机制，使得各个稀奇古怪的隐藏特性需求都可以通过 Feature Flag 来实现，并且此处特性会保持 PR 上的开放，有任何隐藏需求，按照该机制实现即可。 
- [x] 用户鉴权: #2022 这个部分已经有做了一部分 next-auth ，后续计划基于 clerk 提供完善的解决方案。 
- [x] 插件四期：#1310 role=function 到 role -> tools 迁移 & claude / Gemini Functioning Call 能力支持
- [x] App Layout 重构： #2295 ，1) 解决布局文件过于分散的问题，2）利用 NextJS 的平行路由和拦截路由特性，提升产品的用户体验
- [x] 乐观更新改造： #2710 ，解决 serverDB 下请求需要等待的问题，提升体验

由于整个工程量会比以往任何一个单一功能都大，因此各个特性会拆成若干子 PR 进行，全部都完成后就意味着1.0发布。如果社区同学有感兴趣的也可以认领参与。

## 跟进PR 

https://github.com/lobehub/lobe-chat/pull/2889

**Comment by cy948** [comment]
🙋‍♂️有意参与 “用户鉴权” 板块

**Reply by arvinxx** [reply]
好，等我细化 rfc

**Reply by cy948** [reply]
ok

**Comment by Matinal111** [comment]
终于迈向 1.0 版本了，不过我按照 #2556 配置了变量 DATABASE_URL 好像没生效？

**Reply by arvinxx** [reply]
需要搭配另外一个环境变量来使用的，晚些时候我们会把文档补充上
